Update Privacy Statement and Imprint

krusche opened the pull request:
* According to data privacy experts, we should improve the privacy statement and describe the legal basis of data collection in Artemis better based on DSGVO.
* The imprint statement is now part of Artemis. Linking to an external source is not necessary any more. It can be changed by providing a custom html file in the public folder (like the privacy statement)
* The `Accept terms` option is now deactivate by default because acceptance is not necessary in terms of education and examination (as this is the duty of the corresponding university and it does not really makes senes that students can decline to it): Artemis admins who absolute want their users to accept terms can still opt-in in the yml files.

Note: 
* After recommended by the data privacy expert, I adapted the privacy statement from https://dsms.datenschutz.tum.de/privacy-policy
* After recommended by the data privacy expert, I adapted the new imprint from https://dsms.datenschutz.tum.de/imprint

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.

### Steps for Testing
1. Read the privacy statement
2. Read the imprint
3. Try to adapt both files in the public folder
4. Log in to Artemis

Code reviews are highly appreciated as well.

### Screenshots

<img width="420" alt="image" src="https://user-images.githubusercontent.com/744067/115308916-d14d2a80-a16b-11eb-8a7c-503708a6068c.png">

maximilianbergmann reviewed (changes requested):
There is a typo in the privacy statement "Die Rechtsgrundlage für die Verarbeitun".

In the "Zweck" part were grammatic faults - in my opinion it should be:
"Die Rechtsgrundlage für die Verarbeitung Ihrer Daten ergibt sich, soweit nichts anderes angegeben ist, aus Art. 6 Abs. 1 Lit. c (Rechtliche Verpflichtung) der Datenschutz Grundverordnung (DSGVO). Darüberhinaus gelten Art. 61 Bayerisches Hochschulgesetz (BayHschG) sowie die Allgemeine Prüfungs- und Studienordnung (APSO) der TUM und die Bayerische Fernprüfungserprobungsverordnung (BayFEV). Demnach ist es uns erlaubt, die zur Erfüllung einer uns obliegenden Aufgabe erforderlichen Daten zu verarbeiten."

The rest looks good :-)

willergeorg reviewed (changes requested):
A sentence @maximilianbergmann mentioned is in the privacy statement as well and needs to be adapted:

> **Darüber hinaus** gelten Art. 61 Bayerisches Hochschulgesetz (BayHschG) sowie die **die** Allgemeine Prüfungs- und Studienordnung (APSO) der TUM und **die** Bayerische Fernprüfungserprobungsverordnung (BayFEV).

krusche commented:
Thanks, I fixed the issues and also removed a paragraph that was not necessary and was found in the review of the data privacy expert. If there are no more issues, I think we are good to go 👍

FrankeLukas reviewed (approved):
Lgtm.

KiraNickel reviewed (changes requested):
A few more typos in the privacy statement:
In the first paragraph it's still Verarbeitun instead of Verarbeitung.
Privatssphäre -> Privatsphäre. im Rahme -> im Rahmen. im universitäten Umfeld -> im universitären Umfeld.

The rest looks good to me 👍

katjanakosic reviewed (changes requested):
Looked over it with @ArtemTsurikov. Requested changes: Paragraph 1 Line 3 "im Rahmen", P1L3 "Verarbeitun -> Verarbeitung", P2L1 "universitäten -> universitären". Rest looks fine

ArtemTsurikov reviewed (changes requested):
Looked trough it with @katjanakosic. See her change requests

krusche commented:
Thanks, I fixed the mention typos.

KiraNickel reviewed (changes requested):
Found one last typo 😬 
sowie die dem Allgemeine Prüfungs- ...
"dem" needs to go, everything else looks good now

MDK5 commented on src/main/resources/public/content/privacy_statement.html:
```diff
@@ -1,30 +1,124 @@
-    <a href="http://data.europa.eu/eli/reg/2016/679/oj" target="_blank" rel="external noopener noreferrer">Datenschutz Grundverordnung (DSGVO)</a>.
+    Die <a href="https://www.tum.de/" target="_blank" rel="external noopener noreferrer">Technische Universität München (TUM)</a> nimmt den Schutz von personenbezogenen Daten sehr
+    ernst und nutzt eine sichere und verschlüsselte Kommunikation nach bewährten Verfahren und modernsten Technologien
+    (HTTPS mit sicherem Zeritifkat der TUM, TLS 1.3, Strict Transport Security, Forward Secrecy, Same Site Cookie Schutz, etc.) um die Privatsphäre der Nutzer von Artemis bestmöglich zu
```
typo here: etc. should not be preceeded by a ,

MDK5 commented on src/main/resources/public/content/privacy_statement.html:
```diff
@@ -1,30 +1,124 @@
+<h4 id="Technische">Technische Umsetzung</h4>
+<p>
+    Die Webserver von Artemis werden durch die <a href="https://www.rbg.tum.de" target="_blank" rel="external noopener noreferrer">Rechnerbetriebsgruppe</a> der Technischen Universität München betrieben.
+    Die von Ihnen im Rahmen des Besuchs unseres Webauftritts übermittelten personenbezogenen Daten werden daher in unserem Auftrag durch das LRZ verarbeitet:
```
An uninformed reader might not understand what LRZ stands for, a link or explaination would be helpful

NicolasUllmann reviewed (changes requested):
Found 2 typos in the privacy statement as well:
"HTTPS mit sicherem Zeritifkat der TUM" -> "HTTPS mit sicherem Zertifikat der TUM"
"Darüberhinaus" -> "Darüber hinaus"

krusche commented:
Thanks again, I fixed the remaining typos. Let me know if you find even more 👍

KiraNickel reviewed (changes requested):
... und das Bayerische Fernprüfungserprobungsverordnung (BayFEV).
das -> die 
This really is the last one I found 😅 🙈

krusche commented:
> ... und das Bayerische Fernprüfungserprobungsverordnung (BayFEV).
> das -> die
> This really is the last one I found 😅 🙈

thanks, fixed.

MDK5 reviewed (approved):
Tested locally, works as expected; code and text also look good to me.
